### PR TITLE
jarek/5039: autotranslation [WIP]

### PR DESCRIPTION
**Pull request by jaroslawmalekcodete**

### CODE REVIEW

File: beakerx/beakerx/beakerx_autotranslation_server.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+def random_string_generator(size=128):
+    s = ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.ascii_lowercase + string.digits) for _ in
+                range(size))
+    return s
+
+
+def init_env():
+    os.environ["BEAKERX_BASIC_AUTH_USERNAME"] = "beakerx"
+    os.environ["BEAKERX_BASIC_AUTH_PASSWORD"] = random_string_generator()
+    os.environ["BEAKERX_FLASK_SERVER_PORT"] = str(get_free_tcp_port())
```
Comment (scottdraves) [inline beakerx/beakerx/beakerx_autotranslation_server.py]: no need for BEAKERX_BASIC_AUTH_USERNAME just make it constant on the other side
BEAKERX_BASIC_AUTH_PASSWORD → BEAKERX_AUTOTRANSLATION_PASSWORD
BEAKERX_FLASK_SERVER_PORT → BEAKERX_AUTOTRANSLATION_PORT

File: kernel/base/src/main/java/com/twosigma/beakerx/AutotranslationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+  private static String getBasic_auth_username() {
+    return System.getenv("BEAKERX_BASIC_AUTH_USERNAME");
+  }
+
+  private static String getBasic_auth_password() {
+    return System.getenv("BEAKERX_BASIC_AUTH_PASSWORD");
+  }
+
+  private static String flaskServerPort() {
```
Comment (scottdraves) [inline kernel/base/src/main/java/com/twosigma/beakerx/AutotranslationServiceImpl.java]: rename flaskServerPort to autotranslationPort

File: kernel/base/src/main/java/com/twosigma/beakerx/BeakerxClient.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/*
```
Comment (scottdraves) [inline kernel/base/src/main/java/com/twosigma/beakerx/BeakerxClient.java]: rename to BeakerXClient.java

File: kernel/base/src/main/java/com/twosigma/beakerx/BeakerxClientManager.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+/*
```
Comment (scottdraves) [inline kernel/base/src/main/java/com/twosigma/beakerx/BeakerxClientManager.java]: rename BeakerxClientManager.java to BeakerXClientManager.java

File: kernel/base/src/main/java/com/twosigma/beakerx/evaluator/BaseEvaluator.java [hunk 5]
```diff
@@ -100,6 +108,11 @@ public ClassLoader getClassLoaderForImport() {
     return getClassLoader();
   }
 
+  @Override
+  public BeakerxClient getBeakerx() {
```
Comment (scottdraves) [inline kernel/base/src/main/java/com/twosigma/beakerx/evaluator/BaseEvaluator.java]: rename to getBeakerX

**Comment by scottdraves** [comment]
this is awesome!!  it works!

File: beakerx/beakerx/beakerx_autotranslation_server.py (hunk 1 above)
Comment (jaroslawmalekcodete) [reply beakerx/beakerx/beakerx_autotranslation_server.py]: done

File: kernel/base/src/main/java/com/twosigma/beakerx/AutotranslationServiceImpl.java (hunk 2 above)
Comment (jaroslawmalekcodete) [reply kernel/base/src/main/java/com/twosigma/beakerx/AutotranslationServiceImpl.java]: done

File: kernel/base/src/main/java/com/twosigma/beakerx/BeakerxClient.java (hunk 3 above)
Comment (jaroslawmalekcodete) [reply kernel/base/src/main/java/com/twosigma/beakerx/BeakerxClient.java]: done

File: kernel/base/src/main/java/com/twosigma/beakerx/BeakerxClientManager.java (hunk 4 above)
Comment (jaroslawmalekcodete) [reply kernel/base/src/main/java/com/twosigma/beakerx/BeakerxClientManager.java]: done

File: kernel/base/src/main/java/com/twosigma/beakerx/evaluator/BaseEvaluator.java (hunk 5 above)
Comment (jaroslawmalekcodete) [reply kernel/base/src/main/java/com/twosigma/beakerx/evaluator/BaseEvaluator.java]: done

**Comment by scottdraves** [comment]
when in the groovy kernel we run
```beakerx.x  = 1```
and the value gets set in JS, the Proxy object is then sending the value to the server.
But the value came from the server, this is not needed.
Need some kind of back door.
https://github.com/twosigma/beakerx/blob/jarek/5039_autotranslation_v2/js/notebook/src/extension.js#L188

File: js/notebook/src/extension.js [hunk 6]
```diff
@@ -158,17 +158,35 @@ define([
...
+
+      if (!window.beakerx) {
+        var handler = {
+          get: function (obj, prop) {
+            return prop in obj ? obj[prop] : undefined;
+          },
+
+          set: function (obj, prop, value) {
+            obj[prop] = value;
+            var comm = Jupyter.notebook.kernel.comm_manager.new_comm('beaker.autotranslation', null, null, null, utils.uuid());
```
Comment (scottdraves) [inline js/notebook/src/extension.js:178]: create this once and reuse it with each msg.
also, name the channel "beakerx.autotranslation".

Comment (scottdraves) [reply js/notebook/src/extension.js:178]: are all our channels named "beakerx.*"?

Comment (MariuszJurowicz) [reply js/notebook/src/extension.js:178]: nope, but I will change the names from `beaker.*` to `beakerx.*`

**Pull request merged**